src/server/pfs/server/driver.go

```diff
@@ -334,6 +319,31 @@ func (d *driver) listRepo(ctx context.Context, includeAuth bool, repoType string
 	return result, nil
 }
 
+func (d *driver) deleteAllBranchesFromRepos(txnCtx *txncontext.TransactionContext, repos []pfs.RepoInfo, force bool) error {
+	var branchInfos []*pfs.BranchInfo
+	for _, repo := range repos {
+		for _, branch := range repo.Branches {
+			bi, err := d.inspectBranch(txnCtx, branch)
+			if err != nil {
+				return errors.Wrapf(err, "error inspecting branch %s", pretty.CompactPrintBranch(branch))
+			}
+			branchInfos = append(branchInfos, bi)
+		}
+	}
+	// sort ascending provenance
+	sort.Slice(branchInfos, func(i, j int) bool { return len(branchInfos[i].Provenance) < len(branchInfos[j].Provenance) })
+	for i := range branchInfos {
+		// delete branches from most provenance to least, that way if one
+		// branch is provenant on another (such as with stats branches) we
```

Comment:
This is just my lack of understanding but we're moving stats to meta repos too right? 🤞

Reply:
already done in #6185 😏, that was definitely a concern. That change definitely gets rid of the common instances of inter-branch provenance, but I'm not confident there are others (also depends if we nix build pipelines for now)

Reply:
I've amended the comment to better reflect why it's needed